### ISSUE TITLE
hoogle has errors during compilation with GHC7.6, cabal-install 1.22.3.0

### ISSUE THREAD

**Issue by cdepillabout** [ndmitchell/hoogle#115]
Here is the console output of trying to install `hoogle` in a cabal sandbox:

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.6.3
$ cabal --version
cabal-install version 1.22.3.0
using version 1.22.3.0 of the Cabal library
$ happy --version
Happy Version 1.19.0 Copyright (c) 1993-1996 Andy Gill, Simon Marlow (c) 1997-2005 Simon Marlow

Happy is a Yacc for Haskell, and comes with ABSOLUTELY NO WARRANTY.
This program is free software; you can redistribute it and/or modify
it under the terms given in the file 'LICENSE' distributed with
the Happy sources.
$ alex --version
Alex version 3.1.0, (c) 2003 Chris Dornan and Simon Marlow
$ cabal sandbox init
Writing a default package environment file to
/my/path/to/hoogle/cabal.sandbox.config
Creating a new sandbox at /my/path/to/hoogle/.cabal-sandbox
$ cabal install -j hoogle
Resolving dependencies...
Notice: installing into a sandbox located at
/my/path/to/hoogle/.cabal-sandbox
Configuring cmdargs-0.10.12...
Configuring extra-1.1...
Downloading js-flot-0.8.3...
Downloading js-jquery-1.11.2...
Downloading mwc-random-0.13.3.2...
Downloading safe-0.3.8...
Downloading tagsoup-0.13.3...
Building cmdargs-0.10.12...
Building extra-1.1...
Installed extra-1.1
Configuring polyparse-1.11...
Building polyparse-1.11...
Configuring tf-random-0.5...
Installed cmdargs-0.10.12
Building tf-random-0.5...
Configuring uniplate-1.6.12...
Installed tf-random-0.5
Configuring js-jquery-1.11.2...
Installed polyparse-1.11
Building uniplate-1.6.12...
Building js-jquery-1.11.2...
Configuring js-flot-0.8.3...
Installed js-jquery-1.11.2
Building js-flot-0.8.3...
Configuring mwc-random-0.13.3.2...
Installed js-flot-0.8.3
Downloading shake-0.15.1...
Building mwc-random-0.13.3.2...
Configuring safe-0.3.8...
Installed uniplate-1.6.12
Building safe-0.3.8...
Configuring tagsoup-0.13.3...
Installed safe-0.3.8
Building tagsoup-0.13.3...
Installed tagsoup-0.13.3
Configuring QuickCheck-2.8.1...
Building QuickCheck-2.8.1...
Configuring cpphs-1.19...
Installed mwc-random-0.13.3.2
Downloading vector-algorithms-0.6.0.4...
Building cpphs-1.19...
Configuring shake-0.15.1...
Installed QuickCheck-2.8.1
Building shake-0.15.1...
Configuring vector-algorithms-0.6.0.4...
Installed cpphs-1.19
Building vector-algorithms-0.6.0.4...
Installed vector-algorithms-0.6.0.4
Configuring haskell-src-exts-1.16.0.1...
Building haskell-src-exts-1.16.0.1...
Installed shake-0.15.1
Installed haskell-src-exts-1.16.0.1
Downloading hoogle-4.2.40...
Configuring hoogle-4.2.40...
Building hoogle-4.2.40...
Failed to install hoogle-4.2.40
Build log ( /my/path/to/hoogle/.cabal-sandbox/logs/hoogle-4.2.40.log ):
Configuring hoogle-4.2.40...
Building hoogle-4.2.40...
Preprocessing library hoogle-4.2.40...
[ 1 of 50] Compiling General.FMIndex  ( src/General/FMIndex.hs, dist/dist-sandbox-d23c486b/build/General/FMIndex.o )
[ 2 of 50] Compiling General.BurrowsWheeler ( src/General/BurrowsWheeler.hs, dist/dist-sandbox-d23c486b/build/General/BurrowsWheeler.o )
[ 3 of 50] Compiling General.TypeMap  ( src/General/TypeMap.hs, dist/dist-sandbox-d23c486b/build/General/TypeMap.o )
[ 4 of 50] Compiling General.Heap     ( src/General/Heap.hs, dist/dist-sandbox-d23c486b/build/General/Heap.o )
[ 5 of 50] Compiling Paths_hoogle     ( dist/dist-sandbox-d23c486b/build/autogen/Paths_hoogle.hs, dist/dist-sandbox-d23c486b/build/Paths_hoogle.o )
[ 6 of 50] Compiling Hoogle.DataBase2.Type ( src/Hoogle/DataBase2/Type.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase2/Type.o )
[ 7 of 50] Compiling General.Base     ( src/General/Base.hs, dist/dist-sandbox-d23c486b/build/General/Base.o )
[ 8 of 50] Compiling General.System   ( src/General/System.hs, dist/dist-sandbox-d23c486b/build/General/System.o )
[ 9 of 50] Compiling Hoogle.Type.Language ( src/Hoogle/Type/Language.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/Language.o )
[10 of 50] Compiling Hoogle.Score.Type ( src/Hoogle/Score/Type.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Score/Type.o )
[11 of 50] Compiling Hoogle.Score.Scoring ( src/Hoogle/Score/Scoring.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Score/Scoring.o )
[12 of 50] Compiling Hoogle.Score.All ( src/Hoogle/Score/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Score/All.o )
[13 of 50] Compiling General.Util     ( src/General/Util.hs, dist/dist-sandbox-d23c486b/build/General/Util.o )
[14 of 50] Compiling General.Web      ( src/General/Web.hs, dist/dist-sandbox-d23c486b/build/General/Web.o )
[15 of 50] Compiling Hoogle.Store.ReadBuffer ( src/Hoogle/Store/ReadBuffer.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Store/ReadBuffer.o )
[16 of 50] Compiling Hoogle.Store.WriteBuffer ( src/Hoogle/Store/WriteBuffer.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Store/WriteBuffer.o )
[17 of 50] Compiling Hoogle.Store.Type ( src/Hoogle/Store/Type.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Store/Type.o )
[18 of 50] Compiling Hoogle.Store.All ( src/Hoogle/Store/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Store/All.o )
[19 of 50] Compiling Hoogle.Type.TagStr ( src/Hoogle/Type/TagStr.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/TagStr.o )
[20 of 50] Compiling Hoogle.Type.ParseError ( src/Hoogle/Type/ParseError.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/ParseError.o )
[21 of 50] Compiling Hoogle.Type.Docs ( src/Hoogle/Type/Docs.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/Docs.o )
[22 of 50] Compiling Hoogle.Type.TypeSig ( src/Hoogle/Type/TypeSig.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/TypeSig.o )
[23 of 50] Compiling Hoogle.Type.Item ( src/Hoogle/Type/Item.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/Item.o )
[24 of 50] Compiling Hoogle.Type.Result ( src/Hoogle/Type/Result.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/Result.o )
[25 of 50] Compiling Hoogle.Type.All  ( src/Hoogle/Type/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Type/All.o )
[26 of 50] Compiling Hoogle.DataBase2.Str ( src/Hoogle/DataBase2/Str.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase2/Str.o )
[27 of 50] Compiling Hoogle.Language.Haskell ( src/Hoogle/Language/Haskell.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Language/Haskell.o )
[28 of 50] Compiling Hoogle.Query.Type ( src/Hoogle/Query/Type.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Query/Type.o )
[29 of 50] Compiling Hoogle.Query.Parser ( src/Hoogle/Query/Parser.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Query/Parser.o )
[30 of 50] Compiling Hoogle.Query.Render ( src/Hoogle/Query/Render.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Query/Render.o )
[31 of 50] Compiling Hoogle.DataBase.TypeSearch.Binding ( src/Hoogle/DataBase/TypeSearch/Binding.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/Binding.o )
[32 of 50] Compiling Hoogle.DataBase.Aliases ( src/Hoogle/DataBase/Aliases.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Aliases.o )
[33 of 50] Compiling Hoogle.DataBase.Instances ( src/Hoogle/DataBase/Instances.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Instances.o )
[34 of 50] Compiling Hoogle.DataBase.Items ( src/Hoogle/DataBase/Items.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Items.o )
[35 of 50] Compiling Hoogle.DataBase.SubstrSearch ( src/Hoogle/DataBase/SubstrSearch.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/SubstrSearch.o )
[36 of 50] Compiling Hoogle.DataBase.Suggest ( src/Hoogle/DataBase/Suggest.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Suggest.o )
[37 of 50] Compiling Hoogle.DataBase.TypeSearch.EntryInfo ( src/Hoogle/DataBase/TypeSearch/EntryInfo.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/EntryInfo.o )
[38 of 50] Compiling Hoogle.DataBase.TypeSearch.TypeScore ( src/Hoogle/DataBase/TypeSearch/TypeScore.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/TypeScore.o )
[39 of 50] Compiling Hoogle.DataBase.TypeSearch.Result ( src/Hoogle/DataBase/TypeSearch/Result.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/Result.o )
[40 of 50] Compiling Hoogle.DataBase.TypeSearch.Graph ( src/Hoogle/DataBase/TypeSearch/Graph.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/Graph.o )
[41 of 50] Compiling Hoogle.DataBase.TypeSearch.Graphs ( src/Hoogle/DataBase/TypeSearch/Graphs.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/Graphs.o )
[42 of 50] Compiling Hoogle.DataBase.TypeSearch.All ( src/Hoogle/DataBase/TypeSearch/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/TypeSearch/All.o )
[43 of 50] Compiling Hoogle.DataBase.Type ( src/Hoogle/DataBase/Type.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Type.o )
[44 of 50] Compiling Hoogle.DataBase.Serialise ( src/Hoogle/DataBase/Serialise.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/Serialise.o )
[45 of 50] Compiling Hoogle.DataBase.All ( src/Hoogle/DataBase/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/DataBase/All.o )
[46 of 50] Compiling Hoogle.Query.Suggest ( src/Hoogle/Query/Suggest.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Query/Suggest.o )
[47 of 50] Compiling Hoogle.Query.All ( src/Hoogle/Query/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Query/All.o )
[48 of 50] Compiling Hoogle.Search.Results ( src/Hoogle/Search/Results.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Search/Results.o )
[49 of 50] Compiling Hoogle.Search.All ( src/Hoogle/Search/All.hs, dist/dist-sandbox-d23c486b/build/Hoogle/Search/All.o )
[50 of 50] Compiling Hoogle           ( src/Hoogle.hs, dist/dist-sandbox-d23c486b/build/Hoogle.o )
In-place registering hoogle-4.2.40...
Preprocessing executable 'hoogle' for hoogle-4.2.40...
[ 1 of 75] Compiling Test.General     ( src/Test/General.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/General.o )
[ 2 of 75] Compiling Paths_hoogle     ( dist/dist-sandbox-d23c486b/build/autogen/Paths_hoogle.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Paths_hoogle.o )
[ 3 of 75] Compiling Hoogle.DataBase2.Type ( src/Hoogle/DataBase2/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase2/Type.o )
[ 4 of 75] Compiling General.FMIndex  ( src/General/FMIndex.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/FMIndex.o )
[ 5 of 75] Compiling General.BurrowsWheeler ( src/General/BurrowsWheeler.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/BurrowsWheeler.o )
[ 6 of 75] Compiling Test.BWT_FM      ( src/Test/BWT_FM.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/BWT_FM.o )
[ 7 of 75] Compiling General.Base     ( src/General/Base.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/Base.o )
[ 8 of 75] Compiling General.System   ( src/General/System.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/System.o )
[ 9 of 75] Compiling General.Util     ( src/General/Util.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/Util.o )
[10 of 75] Compiling General.Web      ( src/General/Web.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/Web.o )
[11 of 75] Compiling Hoogle.Type.Language ( src/Hoogle/Type/Language.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/Language.o )
[12 of 75] Compiling Hoogle.Score.Type ( src/Hoogle/Score/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Score/Type.o )
[13 of 75] Compiling Hoogle.Score.Scoring ( src/Hoogle/Score/Scoring.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Score/Scoring.o )
[14 of 75] Compiling Hoogle.Store.ReadBuffer ( src/Hoogle/Store/ReadBuffer.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Store/ReadBuffer.o )
[15 of 75] Compiling Hoogle.Store.WriteBuffer ( src/Hoogle/Store/WriteBuffer.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Store/WriteBuffer.o )
[16 of 75] Compiling Hoogle.Store.Type ( src/Hoogle/Store/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Store/Type.o )
[17 of 75] Compiling Hoogle.Store.All ( src/Hoogle/Store/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Store/All.o )
[18 of 75] Compiling Hoogle.Type.TagStr ( src/Hoogle/Type/TagStr.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/TagStr.o )
[19 of 75] Compiling Hoogle.Type.Docs ( src/Hoogle/Type/Docs.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/Docs.o )
[20 of 75] Compiling Hoogle.Type.ParseError ( src/Hoogle/Type/ParseError.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/ParseError.o )
[21 of 75] Compiling Test.Docs        ( src/Test/Docs.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/Docs.o )
[22 of 75] Compiling Hoogle.Type.TypeSig ( src/Hoogle/Type/TypeSig.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/TypeSig.o )
[23 of 75] Compiling Hoogle.Type.Item ( src/Hoogle/Type/Item.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/Item.o )
[24 of 75] Compiling Web.Template     ( src/Web/Template.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Template.o )
[25 of 75] Compiling Web.Page         ( src/Web/Page.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Page.o )
[26 of 75] Compiling Recipe.Haddock   ( src/Recipe/Haddock.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Recipe/Haddock.o )
[27 of 75] Compiling Recipe.Cabal     ( src/Recipe/Cabal.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Recipe/Cabal.o )
[28 of 75] Compiling Console.Log      ( src/Console/Log.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Console/Log.o )
[29 of 75] Compiling Recipe.Keyword   ( src/Recipe/Keyword.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Recipe/Keyword.o )
[30 of 75] Compiling Recipe.Hackage   ( src/Recipe/Hackage.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Recipe/Hackage.o )
[31 of 75] Compiling General.Heap     ( src/General/Heap.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/General/Heap.o )
[32 of 75] Compiling Hoogle.Score.All ( src/Hoogle/Score/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Score/All.o )
[33 of 75] Compiling Hoogle.Type.Result ( src/Hoogle/Type/Result.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/Result.o )
[34 of 75] Compiling Hoogle.Type.All  ( src/Hoogle/Type/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Type/All.o )
[35 of 75] Compiling Hoogle.Language.Haskell ( src/Hoogle/Language/Haskell.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Language/Haskell.o )
[36 of 75] Compiling Hoogle.DataBase.Aliases ( src/Hoogle/DataBase/Aliases.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Aliases.o )
[37 of 75] Compiling Hoogle.DataBase.Instances ( src/Hoogle/DataBase/Instances.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Instances.o )
[38 of 75] Compiling Hoogle.DataBase.Items ( src/Hoogle/DataBase/Items.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Items.o )
[39 of 75] Compiling Hoogle.DataBase.SubstrSearch ( src/Hoogle/DataBase/SubstrSearch.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/SubstrSearch.o )
[40 of 75] Compiling Hoogle.DataBase.Suggest ( src/Hoogle/DataBase/Suggest.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Suggest.o )
[41 of 75] Compiling Hoogle.DataBase.TypeSearch.Binding ( src/Hoogle/DataBase/TypeSearch/Binding.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/Binding.o )
[42 of 75] Compiling Hoogle.DataBase.TypeSearch.EntryInfo ( src/Hoogle/DataBase/TypeSearch/EntryInfo.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/EntryInfo.o )
[43 of 75] Compiling Hoogle.DataBase.TypeSearch.TypeScore ( src/Hoogle/DataBase/TypeSearch/TypeScore.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/TypeScore.o )
[44 of 75] Compiling Hoogle.DataBase.TypeSearch.Result ( src/Hoogle/DataBase/TypeSearch/Result.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/Result.o )
[45 of 75] Compiling Hoogle.DataBase.TypeSearch.Graph ( src/Hoogle/DataBase/TypeSearch/Graph.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/Graph.o )
[46 of 75] Compiling Hoogle.DataBase.TypeSearch.Graphs ( src/Hoogle/DataBase/TypeSearch/Graphs.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/Graphs.o )
[47 of 75] Compiling Hoogle.DataBase.TypeSearch.All ( src/Hoogle/DataBase/TypeSearch/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/TypeSearch/All.o )
[48 of 75] Compiling Hoogle.DataBase.Type ( src/Hoogle/DataBase/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Type.o )
[49 of 75] Compiling Hoogle.DataBase.Serialise ( src/Hoogle/DataBase/Serialise.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/Serialise.o )
[50 of 75] Compiling Hoogle.DataBase2.Str ( src/Hoogle/DataBase2/Str.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase2/Str.o )
[51 of 75] Compiling Hoogle.Query.Type ( src/Hoogle/Query/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Query/Type.o )
[52 of 75] Compiling Hoogle.Query.Parser ( src/Hoogle/Query/Parser.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Query/Parser.o )
[53 of 75] Compiling Hoogle.Query.Render ( src/Hoogle/Query/Render.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Query/Render.o )
[54 of 75] Compiling Hoogle.DataBase.All ( src/Hoogle/DataBase/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/DataBase/All.o )
[55 of 75] Compiling Hoogle.Query.Suggest ( src/Hoogle/Query/Suggest.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Query/Suggest.o )
[56 of 75] Compiling Hoogle.Query.All ( src/Hoogle/Query/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Query/All.o )
[57 of 75] Compiling Hoogle.Search.Results ( src/Hoogle/Search/Results.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Search/Results.o )
[58 of 75] Compiling Hoogle.Search.All ( src/Hoogle/Search/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle/Search/All.o )
[59 of 75] Compiling Test.Parse_TypeSig ( src/Test/Parse_TypeSig.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/Parse_TypeSig.o )
[60 of 75] Compiling Test.Parse_Query ( src/Test/Parse_Query.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/Parse_Query.o )
[61 of 75] Compiling Test.All         ( src/Test/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Test/All.o )
[62 of 75] Compiling Hoogle           ( src/Hoogle.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Hoogle.o )
[63 of 75] Compiling CmdLine.Load     ( src/CmdLine/Load.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/CmdLine/Load.o )
[64 of 75] Compiling CmdLine.Type     ( src/CmdLine/Type.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/CmdLine/Type.o )
[65 of 75] Compiling Console.Rank     ( src/Console/Rank.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Console/Rank.o )
[66 of 75] Compiling CmdLine.All      ( src/CmdLine/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/CmdLine/All.o )
[67 of 75] Compiling Web.Response     ( src/Web/Response.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Response.o )
[68 of 75] Compiling Web.Server       ( src/Web/Server.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Server.o )
src/Web/Server.hs:36:18:
    Not in scope: `setOnException'
    Perhaps you meant one of these:
      `toException' (imported from Control.Exception),
      `onException' (imported from Control.Exception)

src/Web/Server.hs:36:45: Not in scope: `setPort'
cabal: Error: some packages failed to install:
hoogle-4.2.40 failed during the building phase. The exception was:
ExitFailure 1
$
```

**Comment by cdepillabout** [comment]
The above error occurs on all versions all the way back to 4.2.34.  In version 4.2.34, I get the following error:

```
[66 of 75] Compiling Console.Rank     ( src/Console/Rank.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Console/Rank.o )
[67 of 75] Compiling CmdLine.All      ( src/CmdLine/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/CmdLine/All.o )
[68 of 75] Compiling Web.Response     ( src/Web/Response.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Response.o )
[69 of 75] Compiling Web.Server       ( src/Web/Server.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/Server.o )
[70 of 75] Compiling Web.All          ( src/Web/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Web/All.o )
[71 of 75] Compiling Recipe.All       ( src/Recipe/All.hs, dist/dist-sandbox-d23c486b/build/hoogle/hoogle-tmp/Recipe/All.o )

src/Recipe/All.hs:28:12:
    Ambiguous occurrence `CmdLine'
    It could refer to either `Development.Shake.CmdLine',
                             imported from `Development.Shake' at src/Recipe/All.hs:12:1-24
                             (and originally defined in `Development.Shake.Command')
                          or `CmdLine.All.CmdLine',
                             imported from `CmdLine.All' at src/Recipe/All.hs:24:1-18
                             (and originally defined in `CmdLine.Type'
                                at src/CmdLine/Type.hs:19:6-12)

src/Recipe/All.hs:46:10:
    Ambiguous occurrence `CmdLine'
    It could refer to either `Development.Shake.CmdLine',
                             imported from `Development.Shake' at src/Recipe/All.hs:12:1-24
                             (and originally defined in `Development.Shake.Command')
                          or `CmdLine.All.CmdLine',
                             imported from `CmdLine.All' at src/Recipe/All.hs:24:1-18
                             (and originally defined in `CmdLine.Type'
                                at src/CmdLine/Type.hs:19:6-12)

src/Recipe/All.hs:177:9:
    Ambiguous occurrence `CmdLine'
    It could refer to either `Development.Shake.CmdLine',
                             imported from `Development.Shake' at src/Recipe/All.hs:12:1-24
                             (and originally defined in `Development.Shake.Command')
                          or `CmdLine.All.CmdLine',
                             imported from `CmdLine.All' at src/Recipe/All.hs:24:1-18
                             (and originally defined in `CmdLine.Type'
                                at src/CmdLine/Type.hs:19:6-12)
```

**Comment by cdepillabout** [comment]
However, it appears that older versions of hoogle are not able to download the package data successfully?

```
$ hoogle data
2015-05-20 12:00:47 URL:http://hackage.haskell.org/packages/index.tar.gz [9620192/9620192] -> "download/hackage-cabal.tar.gz" [1]
wget: unable to resolve host address ‘old.hackage.haskell.org’
hoogle: Failed to download: http://old.hackage.haskell.org/packages/archive/00-hoogle.tar.gz
```

**Comment by ndmitchell** [comment]
You need the latest version of Hoogle for things to work, as the keyword file moved location. I think the issue is probably that you are using warp < 2.1, and looking at `setOnException`, it was only introduced in 2.1. I've tightened the bounds and released Hoogle 4.2.41. Can you try that?

**Comment by cdepillabout** [comment]
Hoogle 4.2.41 worked!  Thanks for the quick response!

It looks like cabal ended up installing warp-3.0.13.1.
